nucypher/policy/policies.py

```diff
@@ -343,29 +334,42 @@ def enact(self, network_middleware, publish=True) -> dict:
         Assign kfrags to ursulas_on_network, and distribute them via REST,
         populating enacted_arrangements
         """
-        # TODO: #121 - Consider tweaking the order of the enactment steps:
-        # first create the policy on chain and then send the kfrags together with the TX receipt
-
         for arrangement in self.__assign_kfrags():
-            policy_message_kit = arrangement.encrypt_payload_for_ursula()
-
-            response = network_middleware.enact_policy(arrangement.ursula,
-                                                       arrangement.id,
-                                                       policy_message_kit.to_bytes())
+            arrangement_message_kit = arrangement.encrypt_payload_for_ursula()
 
-            if not response:
-                pass  # TODO: Parse response for confirmation.
+            try:
+                response = network_middleware.enact_policy(arrangement.ursula,
+                                                           arrangement.id,
+                                                           arrangement_message_kit.to_bytes())
+            except network_middleware.UnexpectedResponse as e:
+                arrangement.status = e.status
+            else:
+                arrangement.status = response.status_code
 
             # Assuming response is what we hope for.
             self.treasure_map.add_arrangement(arrangement)
 
-        else:  # ...After *all* the arrangements are enacted
+        else:
+            # OK, let's check: if two or more Ursulas claimed we didn't pay,
+            # we need to re-evaulate our situation here.
+            arrangement_statuses = [a.status for a in self._accepted_arrangements]
+            number_of_claims_of_freeloading = sum(status==402 for status in arrangement_statuses)
+
+            if number_of_claims_of_freeloading > 2:
+                raise self.alice.NotEnoughNodes  # TODO: Clean this up and enable re-tries.
```

Comment:
A specific message here about freeloading suspicion...?

Reply:
Later.  We'll work out the Alice side down the road.  I'm trying to keep this PR scoped to fix #121.